### ISSUE TITLE
Immediate crash running v1.27.11 signed release on OpenBSD 7.5

### ISSUE THREAD

**Issue by xorgy** [syncthing/syncthing#9678]
### What happened?

When Syncthing auto-upgraded to 1.27.11 on my OpenBSD 7.5 machine, it failed to start due to dynamic linking against a `syscall` symbol that doesn't exist.

This can be resolved by building Syncthing OpenBSD releases with the OpenBSD Go toolchain. I built the `v1.27.11` tag with go from the OpenBSD packages respository and it works fine.

### Syncthing version

v1.27.11

### Platform & operating system

OpenBSD (7.5)

### Relevant log output

```shell
[REDACTED] 2024/09/03 15:33:45 INFO: Automatic upgrade (current "v1.27.10" < latest "v1.27.11")
[REDACTED] 2024/09/03 15:33:46 WARNING: Automatically upgraded to version "v1.27.11". Restarting in 1 minute.
[REDACTED] 2024/09/03 15:34:46 INFO: Relay listener (dynamic+https://relays.syncthing.net/endpoint) shutting down
[REDACTED] 2024/09/03 15:34:46 INFO: QUIC listener (0.0.0.0:22000) shutting down
[REDACTED] 2024/09/03 15:34:46 INFO: Lost primary connection to REDACTED at REDACTED: Syncthing is being stopped (0 remain)
[REDACTED] 2024/09/03 15:34:46 INFO: Connection to REDACTED at REDACTED closed: Syncthing is being stopped
[REDACTED] 2024/09/03 15:34:46 INFO: TCP listener (0.0.0.0:22000) shutting down
[REDACTED] 2024/09/03 15:34:46 INFO: Exiting
[monitor] 2024/09/03 15:34:46 INFO: Restarting monitor...
syncthing:syncthing: undefined symbol 'syscall'
ld.so: syncthing: lazy binding failed!
fish: Job 1, 'syncthing' terminated by signal SIGKILL (Forced quit)
```

**Comment by calmh** [comment]
We (cross) build with the default/official Go image, and even so the 1.27.11 release didn't really include much changes, merely a bump from Go 1.22.5 to 1.22.6.

**Comment by rasa** [comment]
@xorgy What is the version of go you used to build Syncthing?

Can you try building with go 1.23.0 and see if that's buggy too?

**Comment by xorgy** [comment]
@calmh 
> We (cross) build with the default/official Go image, and even so the 1.27.11 release didn't really include much changes, merely a bump from Go 1.22.5 to 1.22.6.

Yes I noticed that, and curiously even though 1.27.10 was running before, when I run 1.27.10 out of the release tarballs it also fails. I have seen [some potentially related discussion on the OpenBSD mailing lists](<https://www.mail-archive.com/bugs@openbsd.org/msg21313.html>)

**Comment by xorgy** [comment]
> @xorgy What is the version of go you used to build Syncthing?
> 
> Can you try building with go 1.23.0 and see if that's buggy too?

I used the current version of go in the OpenBSD repositories, which is based on 1.22.1, I can take a look and see what the difference might be with upstream go.

**Comment by xorgy** [comment]
@rasa 
> What is the version of go you used to build Syncthing?
> 
> Can you try building with go 1.23.0 and see if that's buggy too?

I gave it a build with Go 1.23.0 on the current OpenBSD snapshot, and it also builds correctly with that.

**Comment by xorgy** [comment]
I believe that package on OpenBSD SNAPSHOT is built from this port in the OpenBSD ports tree `https://github.com/openbsd/ports/tree/master/lang/go`

and when I look at the patches, they seem to mostly (entirely?) add a hardening (Branch Tracking Control Flow Integrity), and otherwise it seems to be a pretty vanilla Go build bootstrapped from 1.22.5, so it's not clear to me why your cross-compiled binaries fail, but my natively compiled ones run fine..

**Comment by xorgy** [comment]
I think that the binary releases are working now on OpenBSD 7.7 so this is moot.
